ratchet_core/src/split/mod.rs

```diff
@@ -275,39 +275,74 @@ where
         matches!(self.close_state.load(Ordering::SeqCst), STATE_OPEN)
     }
 
-    /// Constructs a new text WebSocket message with a payload of `data`.
+    /// Sends a new text WebSocket message with a payload of `data`.
+    ///
+    /// # Cancel safety
+    ///
+    /// This function is not cancellation safe. If the function is called as part of a Tokio or
```

Comment:
I'd just say that it isn't cancellation safe and explain why it isn't in each case. I wouldn't specifically mention the `select!` macros but if you want to you should make clear that you're just using it as an example and there are plenty of other ways cancellation can bite you.